src/state/application/updater.ts

```diff
@@ -45,5 +48,13 @@ export default function Updater(): null {
     dispatch(updateChainId({ chainId }))
   }, [dispatch, debouncedChainId])
 
+  // This will remove selected wallet on disconnect
+  // Needed for some wallets so that on refresh there isn't popup to connect
+  useEffect(() => {
+    if (prevAccount && !account) {
+      dispatch(updateSelectedWallet({ wallet: undefined }))
```

Comment:
Awesome!

Reply:
Could you add a unit-test on it? The change is very helpful, a unit-test will help to protect it from accidental  removing

Reply:
I will add TODO for this